packages/volto/src/components/manage/Widgets/SchemaWidget.jsx

```diff
@@ -142,160 +171,597 @@ const makeFieldsetList = (listOfFieldsets, intl) => {
+  type: 'fieldFactoryInitialData',
+  method: (intl) => ({
+    type: 'array',
+    widget: 'checkbox_group',
+    factory: 'checkbox_group',
+  }),
+});
+
+config.registerUtility({
+  name: 'textarea',
+  type: 'fieldFactoryInitialData',
+  method: (intl) => ({
+    type: 'string',
+    widget: 'textarea',
+    factory: 'textarea',
+  }),
+});
+
+/**
+ * schemaField used for modal form, when editing a field
+ * - based on the factory a set of fields is presented
+ * - fields can be moved to another fieldset
+ * @param {string} factory - the kind of field
+ * @param {Object} intl
+ * @param {*} fieldsets
+ * @param {Boolean} allowEditId
+ * @param {Object} extraFields
+ * @return {Object} - schema
+ */
+const schemaField = (
+  factory,
+  intl,
+  fieldsets,
+  allowEditId,
+  extraFields = {},
+) => {
+  const utility = config.getUtility({
+    name: factory,
+    type: 'fieldFactoryProperties',
+  });
+
+  const properties = utility.method
+    ? utility.method(intl)
+    : {
+        minLength: {
+          type: 'integer',
+          title: intl.formatMessage(messages.minLength),
+        },
+        maxLength: {
+          type: 'integer',
+          title: intl.formatMessage(messages.maxLength),
+        },
+        default: {
+          type: 'string',
+          title: intl.formatMessage(messages.defaultValue),
+        },
+        placeholder: {
+          type: 'string',
+          title: intl.formatMessage(messages.placeholder),
+        },
+      };
+
+  return {
+    fieldsets: [
+      {
+        id: 'default',
+        title: 'default',
+        fields: [
+          ...keys(extraFields),
+          ...(allowEditId ? ['id'] : []),
+          ...['title', 'description', 'parentFieldSet', 'queryParameterName'],
+          ...keys(properties),
+          ...['required'],
+        ],
+      },
+    ],
+    properties: {
+      ...extraFields,
+      ...(allowEditId
+        ? {
+            id: {
               type: 'string',
-              title: intl.formatMessage(messages.choices),
-              widget: 'textarea',
+              title: intl.formatMessage(messages.idTitle),
             },
-          };
-        default:
-          return {
-            minLength: {
-              type: 'integer',
-              title: intl.formatMessage(messages.minLength),
-            },
-            maxLength: {
-              type: 'integer',
-              title: intl.formatMessage(messages.maxLength),
-            },
-          };
-      }
-    })(factory),
-  },
-  required: ['type', 'title'],
-});
+          }
+        : {}),
+      title: {
+        type: 'string',
+        title: intl.formatMessage(messages.title),
+      },
+      description: {
+        type: 'string',
+        widget: 'textarea',
+        title: intl.formatMessage(messages.description),
+      },
+      parentFieldSet: {
+        type: 'string',
+        title: intl.formatMessage(messages.parentFieldSet),
+        choices: makeFieldsetList(fieldsets),
+      },
```

Comment:
This seems to be shown even if there is only one fieldset available, which is distracting.

Reply:
This is not a change in this PR. It is already like this in the main branch.